File: 219. Contains Duplicate II/containsNearbyDuplicate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
const containsNearbyDuplicate = (nums, k) => {
  const indexMap = {};
  for (let i = 0; i < nums.length; i++) {
    if (indexMap[nums[i]] !== undefined && nums[i] === nums[indexMap[nums[i]]] && i - indexMap[nums[i]] <= k) {
      return true;
    }

    indexMap[nums[i]] = i;
  }

  return false;
};


const nums = [1, 2, 3, 1];
const k = 3;
console.log('containsNearbyDuplicate: ', containsNearbyDuplicate(nums, k));

File: 219. Contains Duplicate II/containsNearbyDuplicate.py
from typing import List

class Solution:
  def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
    indexMap = {}
    for i in range(len(nums)):
      if nums[i] in indexMap and nums[i] == nums[indexMap[nums[i]]] and i - indexMap[nums[i]] <= k:
        return True
        
      indexMap[nums[i]] = i
      
    return False


nums = [1, 2, 3, 1]
k = 3
solution = Solution()
print('containsNearbyDuplicate: ', solution.containsNearbyDuplicate(nums, k))
